litellm/llms/anthropic/chat/handler.py

```diff
@@ -376,9 +376,7 @@ async def acompletion_function(
         headers={},
         client: Optional[AsyncHTTPHandler] = None,
     ) -> Union[ModelResponse, CustomStreamWrapper]:
-        async_handler = client or get_async_httpx_client(
```

Comment:
it's not clear to me why this flow needs to be changed. 

What is the problem you're facing/ 

the current get also returns the same httpx client

Reply:
@krrishdholakia our use case is that we're migrating from using the Anthropic SDK directly to using LiteLLM and we want to copy [this behavior](https://gitlab.com/gitlab-org/modelops/applied-ml/code-suggestions/ai-assist/-/blob/21dfeff7f87d566b09131a52ca209981222f13de/ai_gateway/models/base.py) where we're able to supply `event_hooks` to the `httpx.AsyncClient`, which we use for logging low-level details of the request (URL, method, payload). This is why I opted for switching to using the `module_level_aclient`, because then in our initialization code we can say something like 

```py
litellm.module_level_aclient = AsyncHTTPHandler(event_hooks={"request": [hook]})
```

... which will achieve the behavior we want by using that handler and its client for all requests. I saw that on https://github.com/BerriAI/litellm/pull/5567 the streaming anthropic calls were changes from using  `get_async_httpx_client` calls to using `module_level_aclient` ([here](https://github.com/BerriAI/litellm/pull/5567/files#diff-a15e16a66e7453d6d88c194172ca5f414199f0fb6bd1586b19ee8ab8aef734aeR677)), so those are already covered. I did the same for the non-streaming calls to have both scenarios covered.

Reply:
ack on this. @ishaan-jaff i believe you worked on the `get_async_httpx_client` logic, any reason to do that over the module level client?

Reply:
@krrishdholakia I realized that for our use-case we don't need to modify the Anthropic code at all, just having `AsyncHTTPHandler` allow passing in `event_hooks` is sufficient because then we can pass a `client` to the completion call, e.g.:

```py
client = AsyncHTTPHandler(event_hooks={"request": [hook]})

litellm.acompletion("claude-3-5-sonnet-20240620", [{"role": "user", "content": "hi"}], client=client)
```

I've updated this MR to remove all changes to `litellm/llms/anthropic/chat/handler.py`